File: src/main/java/com/example/schoolmanagement/util/EmailFormat.java
package com.example.schoolmanagement.util;

import java.util.regex.Pattern;

/**
 * Kiểm tra định dạng email cơ bản (local@domain), dùng chung cho create/update user.
 */
public final class EmailFormat {

    private static final int MAX_LENGTH = 254;
    /**
     * Mẫu gần với HTML5 / thực tế sử dụng phổ biến; không hỗ trợ đầy đủ RFC 5322.
     */
    private static final Pattern PATTERN = Pattern.compile(
            "^[\\w.+-]+@[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?(?:\\.[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?)+$"
    );

    private EmailFormat() {
    }

    public static boolean isValid(String email) {
        if (email == null || email.isEmpty()) {
            return false;
        }
        if (email.length() > MAX_LENGTH) {
            return false;
        }
        return PATTERN.matcher(email).matches();
    }
}

File: frontend/src/shared/lib/emailFormat.js
/**
 * Khớp logic kiểm tra ở {@code com.example.schoolmanagement.util.EmailFormat} (Java).
 * Giữ đồng bộ khi sửa một trong hai nơi.
 */
const MAX_LENGTH = 254;
const EMAIL_PATTERN =
  /^[\w.+-]+@[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?(?:\.[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?)+$/;

export function isValidEmail(email) {
  if (email == null || typeof email !== 'string') return false;
  const s = email.trim();
  if (!s || s.length > MAX_LENGTH) return false;
  return EMAIL_PATTERN.test(s);
}
